plasmapy/simulation/particle_integrators.py

```diff
@@ -9,7 +9,7 @@
 import numpy as np
 
 
-def boris_push(x, v, B, E, q, m, dt):
+def boris_push(x, v, B, E, q, m, dt, inplace=False):
```

Comment:
On further thought, so as to not make this a breaking change and keep the current behavior the same, let's keep the default as it is. Those jax ideas etc will still work, just with a different default, which is fine, I think.

```suggestion
def boris_push(x, v, B, E, q, m, dt, inplace=True):
```

I'll circle back and modify the docstrings accordingly...